src/config_manager.sh

```diff
@@ -0,0 +1,131 @@
+. $src_script_path/kwio.sh --source-only
+
+# This function handling the save operation of '.config' file. It checks if the
+# '.config' exists and saves it using git (dir.: kw/configs)
+#
+# @force Force option. If it is set and the current name was already saved,
+#        this option will override the '.config' file under the 'name'
+#        specified by '-n' without any message.
+# @name This option specifies a name for a target .config file. This name
+#       represents the access key for .config.
+# @description Description for a config file, de descrition from '-d' flag.
+function save_config_file()
+{
+  local ret=0
+  local force=$1
+  local name=$2
+  local description=$3
+  local current_path=$PWD
+  local -r configs_path="$kw_config_files_path/configs"
+  local -r metadata_dir="metadata"
+  local -r configs_dir="configs"
+
+  if [[ ! -f $current_path/.config ]]; then
+    complain "There's no .config file in the current directory"
+    exit 2 # ENOENT
+  fi
+
+  if [[ ! -d $configs_path ]]; then
+    mkdir $configs_path
+    cd $configs_path
+    git init --quiet $configs_path
+    mkdir $metadata_dir $configs_dir
+  fi
+
+  cd $configs_path
+
+  # Check if the metadata related to .config file already exists
+  if [[ ! -f $metadata_dir/$name ]]; then
+    touch $metadata_dir/$name
+  elif [[ $force != 1 ]]; then
+    if [[ $(ask_yN "$name already exists. Update?") =~ "false" ]]; then
+      complain "Save operation aborted"
+      cd $current_path
+      exit 0
+    fi
+  fi
+
+  if [[ ! -z $description ]]; then
+    echo $description > $configs_path/$metadata_dir/$name
+  fi
+
+  cp $current_path/.config $configs_path/$configs_dir/$name
+  git add .
+  git commit -m "New config file added: $USER - $(date)" > /dev/null 2>&1
+
+  if [[ "$?" == 1 ]]; then
+    warning "Warning: $name: there's nothing new in this file"
+  else
+    success "Saved $name"
+  fi
+
+  cd $current_path
+}
+
+function list_configs()
+{
+  local -r configs_path="$kw_config_files_path/configs"
+  local -r metadata_dir="metadata"
+  local -r configs_dir="configs"
+
+  if [[ ! -d $configs_path ]]; then
+    say "There's no tracked .config file"
+    exit 0
+  fi
+
+  echo -e "Name\t\tDescription"
+  echo -e "----\t\t------------"
+  for filename in $configs_path/$metadata_dir/*; do
+    local name=$(basename $filename)
+    local content=$(cat $filename)
+    echo -n $name
+    echo -e "\t\t$content"
+  done
+}
+
+# Support function for execute_config_manager()
+function get_option_value()
+{
```

Comment:
I understood this function by it's use. But by itself, it's really hard to understand what it does. Do you think it would be to bad to remove it and use this, instead, in the only place where it is needed:
```
shift && name_config=$(if [[ $1 = '-n' ]]; then; echo $2; fi)
```